cudax/include/cuda/experimental/__memory_resource/any_resource.cuh

```diff
@@ -95,18 +233,34 @@ public:
           nullptr, &_CUDA_VMR::__alloc_vtable<_Alloc_type, _CUDA_VMR::_WrapperType::_Owning, __resource_t>)
       , __vtable(__vtable::template _Create<__resource_t>())
   {
-    if constexpr (_CUDA_VMR::_IsSmall<__resource_t>())
-    {
-      ::new (static_cast<void*>(this->__object.__buf_)) __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
-    }
-    else
-    {
-      this->__object.__ptr_ = new __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
-    }
+    static_assert(_CUDA_VMR::_IsSmall<__resource_t>());
+    ::new (static_cast<void*>(this->__object.__buf_)) __resource_t(_CUDA_VSTD::forward<_Resource>(__res));
```

Comment:
Suggestion: I know that was preexisting but we should use `cuda::std::__construct_at`